openedx/core/djangoapps/discussions/migrations/0001_initial.py

```diff
@@ -0,0 +1,32 @@
+# Generated by Django 2.2.14 on 2020-07-22 11:06
+
+from django.db import migrations, models
+import opaque_keys.edx.django.models
+
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = [
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='ExternalDiscussionsIdMapping',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('context_key', opaque_keys.edx.django.models.LearningContextKeyField(db_index=True, max_length=255)),
+                ('usage_key', opaque_keys.edx.django.models.UsageKeyField(db_index=True, max_length=255)),
+                ('external_discussion_id', models.CharField(db_index=True, max_length=255)),
```

Comment:
Do we have a sense of what any of these external ids actually look like in practice yet?
The field is a string, so we're flexible there and the length is reasonably long (and could be increased?), so I think we're flexible enough; I'm just curious :)

Reply:
I do not personally have much context on what these would look like. My understanding is that they would be relatively arbitrary slugs that reference the id of the discussion from whichever external tool generates/hosts/handles the discussion itself. @xitij2000 might have a bit more context/information here

Reply:
It's currently unknown, but for discourse it might be topic/category id, so it's an alphanumeric sequence usually. I don't expect it to ever be that long.